src/libs/ReportUtils.js

```diff
@@ -1324,12 +1324,27 @@ function getLastVisibleMessage(reportID, actionsToMerge = {}) {
 }
 
 /**
- * Determines if a report has an IOU that is waiting for an action from the current user (either Pay or Add a credit bank account)
+ * Checks if a report is an open task report assigned to current user.
+ *
+ * @param {Object} report
+ * @param {Object} parentReportAction - The parent report action of the report (Used to check if the task has been canceled)
+ * @returns {Boolean}
+ */
+function isWaitingForTaskCompleteFromAssignee(report, parentReportAction = {}) {
+    return isTaskReport(report) && isReportManager(report) && isOpenTaskReport(report, parentReportAction);
+}
+
+/**
+ * Determines if a report should show a GBR (green dot) in the LHN. This can happen when the report: 
+    - is unread and the user was mentioned in one of the unread comments
+    - is for an outstanding task waiting on the user
+    - has an outstanding child money request that is waiting for an action from the current user (e.g. pay, approve, add bank account)
  *
  * @param {Object} report (chatReport or iouReport)
+ * @param {Object} parentReportAction (the report action the current report is a thread of)
  * @returns {boolean}
  */
-function isWaitingForIOUActionFromCurrentUser(report) {
+function shouldShowGBR(report, parentReportAction = {}) {
```

Comment:
Is this a `report` or `optionItem`? They are have some of the same fields, but not the same thing.

Reply:
Hmm interesting. Right now this function seems to be called using both reports and optionItems. What is the significance of them being different? And is that a problem?

Reply:
>What is the significance of them being different?

They are just different. One is used for the LHN options and one is a report object used for other things besides the LHN.

>And is that a problem?

Could be different ways to interpret your question.  If you mean whether it's a problem to have a misleading or generic argument name then the answer is probably "it depends". We are getting stricter about this I believe. Migrating to TypeScript and all that. So, maybe if we expect this to be a report then call it a report. If it's an option item then call it that.

Reply:
Ok interesting. Right now, we use this function in places where it is called with optionItems and in places where it is called with reports. 

For example, in `OptionRowLHN` we call it with an optionItem. In `SidebarUtils` we call it with a report. And then in `ReportUtils` we call it in the `shouldReportBeInOptionList` function with reports, but that function is used by `OptionListUtils` and `SidebarUtils` to get options 🤔 

So there seems to be a lot of cross-contamination in terms of functions used with optionItems and reports. So I'm trying to wrap my head around what the best approach would be here. Given that sometimes we want to know if a report requires attention from the current user. And sometimes we want to know that for an optionItem. But the information we'd use to figure that out is the same for both of them. 

So it seems to me like the options are: 
1. leave what we have today 
2. have similar `requiresAttentionFromCurrentUser` functions in `ReportUtils` and `OptionsListUtils` 
3. have a `ReportOrOptionItem` typescript type
4. something else I'm not thinking of

Do you have a strong opinion or disagree on those points?

Reply:
Union types seem fine. I'd go with option 3 on your list.

Reply:
sounds good! updated it to be `option`.